src/machines/workflow_machines.rs

```diff
@@ -479,6 +481,17 @@ impl WorkflowMachines {
             }
         }
 
+        let first_event_id = match events.first() {
+            Some(event) => event.event_id,
+            None => 0,
+        };
+        // Workflow has been evicted, but we've received partial history from the server.
+        // Need to reset sticky and trigger another poll.
+        if self.current_started_event_id == 0 && first_event_id != 1 && !events.is_empty() {
+            debug!("Workflow cache miss. Resetting sticky and triggering another poll.");
```

Comment:
I'd just say cache miss here. Technically at this level it's not known what the caller will do with this info so the log line could be slightly misleading

Reply:
Done